src/NLog/Targets/MailTarget.cs

```diff
@@ -284,13 +307,26 @@ protected override void Write(AsyncLogEventInfo[] logEvents)
         /// </summary>
         protected override void InitializeTarget()
         {
-
+            InitializeFrom();
             CheckRequiredParameters();
 
             base.InitializeTarget();
         }
 
         /// <summary>
+        /// In contrary to other settings, System.Net.Mail.SmtpClient doesn't read the 'From' attribute from the system.net/mailSettings/smtp section in the config file.
+        /// Thus, when UseSystemNetMailSettings is enabled we have to read the configuration section of system.net/mailSettings/smtp to initialize the 'From' address.
+        /// It will do so only if the 'From' attribute in system.net/mailSettings/smtp is not empty.
+        ///  </summary>
+        private void InitializeFrom()
+        {
+            if (UseSystemNetMailSettings)
+            {
+                From = string.IsNullOrEmpty(MailSettings.From) ? From : MailSettings.From;
```

Comment:
This is correct, but reads a bit strange. If `MailSettings.From == null`, then we do `From = From`. Please rewrite to a `if`